### PR TITLE
feat(gaussdb/opengauss): add instances data source

### LINKED ISSUE #1449: [GaussDB] Add gaussdb_opengauss_instances data source support
I'd like to add support for gaussdb opengauss instances data source.

### PR DESCRIPTION
**Pull request by niuzhenguo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1449 

**Special notes for your reviewer**:
Customers specifically require `instances` data source instance of `instance`

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/gaussdb' TESTARGS='-run=TestAccSomethingV0_basic'
...
=== RUN   TestAccOpenGaussInstancesDataSource_basic
--- PASS: TestAccOpenGaussInstancesDataSource_basic (1296.37s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   1296.407s
```

### CODE REVIEW

File: docs/data-sources/gaussdb_opengauss_instances.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - Indicates the ID of the instance.
+
+The `instances` block supports:
+
+* `region` - The region of the instance.
```
Comment (chengxiangdong) [inline docs/data-sources/gaussdb_opengauss_instances.md:36]: Missing instance id

Comment (niuzhenguo) [reply docs/data-sources/gaussdb_opengauss_instances.md:36]: done

**Comment by chengxiangdong** [comment]
LGTM

File: docs/data-sources/gaussdb_opengauss_instances.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+* `vpc_id` - (Optional, String) Specifies the VPC ID.
+
+* `subnet_id` - (Optional, String) Specifies the network ID of a subnet.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - Indicates the ID of the instances.
```
Comment (Jason-Zhang9309) [inline docs/data-sources/gaussdb_opengauss_instances.md]: What' the difference between this `id` and the `id` in instances block?

Comment (niuzhenguo) [reply docs/data-sources/gaussdb_opengauss_instances.md]: id within this block is separated instance id, the id outside is the data source id which is hashcode of the list ids

**Review by Jason-Zhang9309 — COMMENTED** [review]
LGTM

File: huaweicloud/services/acceptance/gaussdb/data_source_huaweicloud_gaussdb_opengauss_instances_test.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  cidr = "192.168.0.0/16"
+}
+
+resource "huaweicloud_vpc_subnet" "test" {
+  name          = "%s"
+  cidr          = "192.168.0.0/16"
+  gateway_ip    = "192.168.0.1"
+
+  primary_dns   = "100.125.1.250"
+  secondary_dns = "100.125.21.250"
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/gaussdb/data_source_huaweicloud_gaussdb_opengauss_instances_test.go]: please use the **default dns** as we can not guarantee that the region will always be cn-north-4.

Comment (niuzhenguo) [reply huaweicloud/services/acceptance/gaussdb/data_source_huaweicloud_gaussdb_opengauss_instances_test.go]: done

**Pull request merged**